examples/sunmatrix/band/test_fsunmatrix_band_mod.f90

```diff
@@ -118,21 +119,19 @@ integer(C_INT) function unit_tests() result(fails)
 
     ! Fill identity matrix
     Idata => FSUNBandMatrix_Data(I)
-    do jj = 0, N-1
-      colj => FSUNBandMatrix_Column(I, jj)
-      colj(1) = ONE
+    do jj = 1, N
+      Idata(jj) = ONE
     end do
 
     ! Fill A matrix
     smu   = FSUNBandMatrix_StoredUpperBandwidth(A)
     Adata => FSUNBandMatrix_Data(A)
     do jj = 0, N-1
-      offset = smu-mu+1 + jj*(smu+ml+1)
-      colj(-mu:ml) => Adata(offset:mu+ml)
+      offset = smu-mu + jj*(smu+ml+1)
       jstart = merge(-jj, -mu, jj < mu)
       jend = merge(N-1-jj, ml, jj > N-1-ml)
       do ii = jstart, jend
-        colj(ii) = jj - ii
+        Adata(offset+mu+ml+ii) = jj - ii
```

Comment:
The updating indexing does not seem to produce the same matrix as the C test and results in a floating point exception in the matvec.

Reply:
My mistake this should haven been `Adata(offset+ii) = (jj - 1) * ii`. I'm not sure how the old indexing was working. `offset+mu+ml` is not the diagonal entry for column `jj` when `offset = smu-mu + jj*(smu+ml+1)`.

Reply:
I think all we need to do is keep the indexing Ive written, but change `offset` back to `offset = smu-mu+1 + jj*(smu+ml+1)` and then access it as `Adata(offset+mu+ii)`.  The current case with `offset+mu+ml+ii` only works because `mu` and `ml` are both 1 in the test.

Reply:
> My mistake this should haven been `Adata(offset+ii) = (jj - 1) - ii`. I'm not sure how the old indexing was working. `offset+mu+ml` is not the diagonal entry for column `jj` when `offset = smu-mu + jj*(smu+ml+1)`.

The old indexing was working because `mu` and `ml` were 1.  I think now with my update the indexing you're doing is equivalent to my fixed version.

Reply:
Sounds good, but I'd suggest using `offset = smu + 1 + jj * (smu + ml + 1)` and `Adata(offset + ii)` so the offset is the diagonal entry.

Reply:
Ill go with your fixed indexing scheme since it makes it `offset+ii` instead of `offset+mu+ii`.